crates/iceberg/src/writer/mod.rs

```diff
@@ -26,23 +26,58 @@
 //! 2. IcebergWriter: Focus on the logical format of iceberg table. It will write the data using the FileWriter finally.
 //!
 //! # Simple example for data file writer:
```

Comment:
Should we put this example under the `examples` crate?

https://github.com/apache/iceberg-rust/tree/main/crates/examples/src